openmdao/components/submodel_comp.py

```diff
@@ -382,7 +410,7 @@ def _setup_var_data(self):
             p.model.add_constraint(prom_name)
 
         # setup again to compute coloring
-        p.set_solver_print(-1)
+        p.set_solver_print(**self._subprob_solver_print_options)
```

Comment:
Don't bother with this line, just remove the previous line and let whatever settings the user had previously applied remain.